### DISCUSSION TITLE
MPO representation in tensorcircuit

### DISCUSSION THREAD

**Discussion by gaox-qd** [tencent-quantum-lab/tensorcircuit#17] [General]
Hi all, excellent work!
The construction of MPO representation for a general Hamiltonian should be very interesting, is it possible to get the MPO currently for a Hamiltonian besides FiniteTFI as shown in the example?

**Comment by refraction-ray** [comment]
Good question. Currently, we have utility function in tensorcircuit as ``tc.quantum.quimb2qop`` and  ``tc.quantum.tn2qop`` which support MPO transformation from quimb package and tensornetwork package. For quimb package, you can build **general spin Hamiltonian with NN interaction**, see https://github.com/jcmgray/quimb/blob/develop/quimb/tensor/tensor_gen.py#L2236. And for the utility transformation function usage, you can check test cases in https://github.com/tencent-quantum-lab/tensorcircuit/blob/master/tests/test_quantum.py#L364-L409.
